### PR TITLE
[Improvement] Develop mocks platform

### PR DESCRIPTION
**Pull request by philschanely**
## Description

This PR attempts to provide a home for mocks within the Docs site.

## Background

Several UXD have been participating in a trial of a workflow for how we may hand off work from and to Product teams in need of new or revised features in Sage. Part of this process has been to create a Proof of Concept (PoC) mock of the feature requested by a Product team in order to a) test what is currently possible in Sage, b) deduce what is needed in order to accomplish the feature, c) assist the Product team for both in flight work on the feature and for final handoff of the completed aspects. During this trial we used the Sandbox area to create the mocks and left them in Draft PRs.

This PR drafts a more permanent home for such mocks. Some notes:

- The area drafted here resides at `/pages/mocks` for a landing page with an index of active mocks and individual mocks may be accessed and shared via `/pages/mock/[mock-alias]`. These are currently not on the main navigation and would need to be provided to the respective team at handoff.
- The individual mock displays a preview of the mocked feature along with links to the effort's milestone in Github where all individual issues related tot he effort are organized and prioritized. We also provide links to mock code on Github to which product teams may refer. This currently includes Rails mocks partials, a Rails view helper, Rails-related JS, Custom SCSS. We also optionally link to a Storybook mock if one is provided. 
- All mock-related code is kept out of the distributed bundle they do not increase the package size.
- This includes an archive setting that allows us to continue to groom the available mocks.

A mock may include any of the following:

- An entry in `sage_mocks` within `docs/app/helpers/mocks_helper.rb`
- Rails partials organized in a folder within `docs/app/views/mocks/[alias]`; the primary should be named `_main`.
- Rails helper named `[alias]_helper.rb` within `docs/app/helpers/mocks` for oranizing data the partials use.
- Rails JS scripts organized in a folder within `docs/lib/sage-frontend/javascript/docs/mocks/[alias]`. Main file should be `index.js` but can include other files as needed.
- Custom SCSS placed in `docs/lib/sage-frontend/stylesheets/docs/mocks/_[alias].scss`. Hopefully this will have a decreased need over time.
- React mocks through Storybook story file in `packages/sage-react/lib/mocks/[alias].story.jsx`.

Future could include:

- A generate command to quickly generate all necessary files.
- A "last updated" timestamp on the mock for teams to continue to track status.
- Smarter Github linking so that the current branch is used rather than always `main`.
- Smoother integration with Storybook/React mocks including possibly rendering the Story inline in a tab.

## Screenshots

Main docs page

Screen Shot 2021-11-05 at 5 11 08 PM

Individual mock

Screen Shot 2021-11-05 at 5 10 47 PM

Mock source code dropdown

Screen Shot 2021-11-05 at 5 10 53 PM

## Testing in `sage-lib`

See http://localhost:4000/pages/mocks locally.

Note that source code links have the `main` branch hard coded for now. Future we can look into getting the current branch name in instead. Should work correctly after deploying.

## Testing in `kajabi-products`

1. (LOW) Docs-only addition of a structure for showing Mocks for product teams.

### CODE REVIEW

File: docs/app/helpers/mocks_helper.rb [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        status: ARCHIVED,
+        team: "UXD",
+        storybook_url: nil,
+      },
+      {
+        alias: "contact_profile",
+        milestone_id: 21,
+        name: "Contact Profile",
+        status: DOING,
+        team: "Manage",
```
Comment (monicawheeler) [inline docs/app/helpers/mocks_helper.rb:32]: A small thing here, but could it read "Team: Manage" as the value? 
Screen Shot 2021-11-08 at 10 51 34 AM

File: docs/lib/sage-frontend/javascript/docs/mocks/contact_profile/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+console.log('contact profile js');
```
Comment (monicawheeler) [inline docs/lib/sage-frontend/javascript/docs/mocks/contact_profile/index.js:1]: Do we still need this log?

File: docs/lib/sage-frontend/javascript/docs/mocks/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+require('./contact_profile')
+
+console.log('mocks js loaded');
```
Comment (monicawheeler) [inline docs/lib/sage-frontend/javascript/docs/mocks/index.js:3]: Do we still need this log?

**Review by monicawheeler — COMMENTED** [review]
This looks great! Really nice to have a dedicated place to demo and proof these designs! 
Some minor comments added.

File: docs/app/views/mocks/archives/test/_main.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<p>Main view!!</p>
```
[the remark this answers is not in the document]
Comment (philschanely) [reply docs/app/views/mocks/archives/test/_main.html.erb:1]: No; I can def clean up this test mock. In fact, event he contact profile one will need to sync up with the official mock too.

**Review by joedakroub — APPROVED** [review]
This is looking terrific Phil; very nicely done. 👍🏼 

I am wondering if providing a date or timestamp for every card would be useful? 

In addition, I wonder if adding categories might be something we might want to add or plan for in a future update. Just a couple of thoughts, non-blocking.

File: docs/app/views/mocks/_heading.html.erb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    text: "Back",
+    url: pages_mocks_path(),
+  }]} %>
+  <h1>
+    <%= mock[:name] %>
+    <%= sage_component SageProperty, {
+      icon: "users-alt",
+      value: "#{mock[:team]}",
+      html_attributes: {
+        "data-js-tooltip": "Who this is for",
```
[the remark this answers is not in the document]
Comment (philschanely) [reply docs/app/views/mocks/_heading.html.erb:13]: Got it!

File: docs/app/views/mocks/_heading.html.erb [hunk 6]
```diff
@@ -0,0 +1,56 @@
+
```
[the remark this answers is not in the document]
Comment (philschanely) [reply docs/app/views/mocks/_heading.html.erb:1]: Nope. Good catch!

**Comment by philschanely** [comment]
@monicawheeler I've updated based on your comments. Mind having another look?

File: docs/app/views/mocks/_heading.html.erb [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    text: "Back",
+    url: pages_mocks_path(),
+  }]} %>
+  <h1>
+    <%= mock[:name] %>
+    <%= sage_component SageProperty, {
+      icon: "users-alt",
+      value: "#{mock[:team]}",
+      html_attributes: {
+        "data-js-tooltip": "Built for #{mock[:team]} team",
```
Comment (monicawheeler) [inline docs/app/views/mocks/_heading.html.erb:12]: 😍

File: docs/lib/sage-frontend/stylesheets/docs/mocks/_contact_profile.scss [hunk 8]
```diff
@@ -0,0 +1 @@
+// Cusotm styles go here (as needed)
```
Comment (monicawheeler) [inline docs/lib/sage-frontend/stylesheets/docs/mocks/_contact_profile.scss:1]: Minor spelling fix needed 🙈

**Review by monicawheeler — APPROVED** [review]
Nice! Thanks for collaborating here Phil!

**Comment by philschanely** [comment]
@joedakroub noted for future work. I agree some form of "last updated" is going to be very relevant. I would like to leave the categorization to sort out a bit more as things evolve.

**Pull request merged**